ansible/install_dependencies.sh

```diff
@@ -1,33 +1,70 @@
 #!/bin/bash
 
-echo Enabling universe repository
-sudo add-apt-repository universe
-
 echo Enabling docker repository
-curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
-sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+sudo mkdir -p /etc/apt/keyrings
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg --yes
+
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+  $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
 
 # Update local repositories
 echo Updating local repositories
 sudo apt-get update
 
-# Install ansible dependencies
+# Install dependencies
 echo Installing dependencies
-sudo apt-get install -y apt-transport-https ca-certificates curl gnupg-agent software-properties-common sshpass pv gawk
+sudo apt-get install -y make curl wget libltdl7 libseccomp2 libffi-dev gawk apt-transport-https ca-certificates curl gnupg gnupg-agent lsb-release software-properties-common sshpass pv
+
+# Install python dependencies
+echo Installing Python dependencies
+sudo apt-get install -y python3-distutils
+sudo apt-get install -y python3-pip
+python3 -m pip install -U pip setuptools
+
+# Install ansible if not present
+if [ "`which ansible`" != ""  ]; then
+    echo ansible already installed, skipping.
+else
+    echo Installing ansible
+    python3 -m pip install --user ansible
+fi
 
 # Install docker if not present
-if hash docker 2>/dev/null; then
+if [ "`which docker`" != ""  ]; then
     echo Docker already installed, skipping.
 else
     echo Installing docker
-    sudo apt-get install -y docker-ce docker-ce-cli containerd.io
+    sudo apt-get install -y docker-ce docker-ce-cli containerd.io docker-compose-plugin
+    sudo adduser $USER vboxsf
 fi
 
-# Install docker compose
-echo Installing docker-compose
-sudo curl -L "https://github.com/docker/compose/releases/download/1.25.5/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
-sudo chmod +x /usr/local/bin/docker-compose
-sudo ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
+# Install Go if not present
+if [ "`which go`" != "" ]; then
+    IFS=' '
+    v=`go version | { read _ _ v _; echo ${v#go}; }`
+    IFS='.'
+    read -ra v <<< "$v"
+    if (( ${v[0]} == 1 && ${v[1]} >= 17 )); then
+        echo Go 1.17+ already installed, skipping.
+    else
+        echo Removing existing Go installation
```

Comment:
Removing the existing Go installation should be confirmed with user. 
The user may have installed multiple versions of Go for other usage.